test/microbench/org/apache/cassandra/test/microbench/sstable/SSTableReaderBench.java

```diff
@@ -0,0 +1,209 @@
+
+    @Param({ "BIG", "BTI" })
+    public String formatName;
+    private ColumnFamilyStore table;
+    private LifecycleTransaction txn;
+
+    @Setup
+    public void setup()
+    {
+        Keyspace ks = prepareMetadata();
+        table = ks.getColumnFamilyStore(tableName);
+        pkeys = prepareDecoratedKeys(0, P_KEYS, KEY_SIZE);
+        nonpkeys = prepareDecoratedKeys(P_KEYS, P_KEYS * 2, KEY_SIZE);
+        ckeys = prepareBuffers(0, C_KEYS, KEY_SIZE);
+    }
+
+    @Setup(Level.Iteration)
+    public void setupIteration() throws Exception
+    {
+        txn = LifecycleTransaction.offline(OperationType.WRITE);
+        sstr = prepareReader(getFormat(formatName), table, txn);
+    }
+
+    private int nextIdx() {
+        idx += step;
+        if (idx >= P_KEYS)
+            idx -= P_KEYS;
+        return idx;
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.Throughput)
+    public void getEQPosition()
+    {
+        sstr.getPosition(pkeys[nextIdx()], SSTableReader.Operator.EQ);
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.Throughput)
+    public void getEQPositionNonExisting()
+    {
+        sstr.getPosition(nonpkeys[nextIdx()], SSTableReader.Operator.EQ);
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.Throughput)
+    public void getGTPosition()
+    {
+        sstr.getPosition(pkeys[nextIdx()], SSTableReader.Operator.GT);
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.Throughput)
+    public void getGTPositionNonExisting()
+    {
+        sstr.getPosition(nonpkeys[nextIdx()], SSTableReader.Operator.GT);
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.AverageTime)
+    public void iterateOverAllKeys() throws Exception
+    {
+        try (PartitionIndexIterator it = sstr.allKeysIterator())
+        {
+            while (!it.isExhausted()) it.advance();
+        }
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.AverageTime)
+    public void fullScanTest()
+    {
+        try (ISSTableScanner scanner = sstr.getScanner())
+        {
+            while (scanner.hasNext())
+            {
+                UnfilteredRowIterator rowIt = scanner.next();
+                while (rowIt.hasNext())
+                {
+                    rowIt.next();
+                }
+            }
+        }
+    }
+
+    @TearDown(Level.Iteration)
+    public void tearDownIteration()
+    {
+        sstr.selfRef().release();
+        txn.finish();
+        txn.close();
+        FileUtils.deleteRecursive(sstr.descriptor.directory);
+    }
+
+    private SSTableReader prepareReader(SSTableFormat format, ColumnFamilyStore table, LifecycleTransaction txn) throws Exception
+    {
+        try (SSTableWriter tableWriter = createWriter(table, format, txn))
+        {
+            for (int i = 0; i < P_KEYS; i++)
+            {
+                UpdateBuilder builder = UpdateBuilder.create(table.metadata(), pkeys[i].getKey().duplicate()).withTimestamp(1);
+                if (table.metadata().clusteringColumns().isEmpty())
```

Comment:
I do not know if this check is expensive... if so then maybe it should be extracted to some local var before `for i` ?

Reply:
it is used in setup phase, not in the benchmark itself